src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -528,3 +570,60 @@ object IndexLogEntry {
 
   def schemaString(schema: StructType): String = schema.json
 }
+
+/**
+ * Provides functionality to generate unique file ids for files.
+ */
+class FileIdTracker {
+  private var maxId: Long = -1L
+
+  // Combination of file properties, used as key, to identify a
+  // unique file for which an id is generated.
+  type key = (
+    String, // Full path.
+      Long, // Size.
+      Long  // Modified time.
+    )
+  private val fileToIdMap: mutable.HashMap[key, Long] = mutable.HashMap()
+
+  def getMaxFileId: Long = maxId
+
+  def getFileToIdMap: HashMap[key, Long] = fileToIdMap
+
+  def getFileId(path: String, size: Long, modifiedTime: Long): Option[Long] =
+    fileToIdMap.get((path, size, modifiedTime))
+
+  def setSizeHint(size: Int): Unit = fileToIdMap.sizeHint(size)
+
+  // FileInfo's 'name' contains the full path to the file.
+  def addFileInfo(files: Set[FileInfo]): Unit = {
+    setSizeHint(files.size)
+    files.foreach { f =>
+    val key = (f.name, f.size, f.modifiedTime)
+    fileToIdMap.get(key) match {
+      case Some(id) =>
+        assert(id == f.id)
```

Comment:
This should throw an exception. Can you please add the description of this function simliar to `addFile` below?